src/main/java/com/tabnine/binary/fetch/BinaryVersionFetcher.java

```diff
@@ -27,6 +28,12 @@ public BinaryVersionFetcher(LocalBinaryVersions localBinaryVersions, BinaryRemot
      * @throws SecurityException, NoExistingBinaryException if something went wrong
      */
     public String fetchBinary() throws NoValidBinaryToRunException {
+        Optional<BinaryVersion> bootstrappedVersion = BootstrapperSupport.bootstrap_version(localBinaryVersions, binaryRemoteSource, binaryDownloader);
```

Comment:
This being static is interfering with [Dependency Inversion Principle](https://en.wikipedia.org/wiki/Dependency_inversion_principle). So make it a class and consume an instance through the constructor. Exceptions are everywhere here.

Reply:
this is all temporary until we are going to remove the other code. . I don't think it is worth the time effort as the bootstrapper support code will be inlined to the existing function

Reply:
There is no such thing as a temporary code. Everything in master should be addressed as production code.